src/main/java/org/rocstreaming/roctoolkit/Receiver.java

```diff
@@ -125,7 +125,22 @@
  * @see ReceiverConfig
  * @see java.lang.AutoCloseable
  */
-public class Receiver extends NativeObject implements AutoCloseable {
+public class Receiver extends NativeObject {
+
+    /**
+     * Validate receiver constructor parameters and open a new receiver if validation is successful.
+     *
+     * @param context                       should point to an opened context.
+     * @param config                        should point to an initialized config.
+     *
+     * @return                              the native roc receiver pointer.
+     * @throws IllegalArgumentException     if the arguments are invalid.
+     * @throws Exception                    if an error occured when creating the receiver.
+     */
+    private static long validate(Context context, ReceiverConfig config) throws IllegalArgumentException, Exception {
```

Comment:
Maybe better name these methods `validateAndOpen()` or `tryOpen()` or something like this? It was surprising to me that validates calls open.

Reply:
Sure I'll rename these to `tryOpen`.

Reply:
renamed in https://github.com/roc-streaming/roc-java/pull/40